accord-core/src/test/java/accord/verify/ElleVerifier.java

```diff
@@ -0,0 +1,384 @@
+                        return value;
+                default:
+                    throw new IndexOutOfBoundsException();
+            }
+        }
+
+        @Override
+        public int size()
+        {
+            return value == null ? 2 : 3;
+        }
+    }
+
+    private static class Read extends Action
+    {
+        protected Read(int key, int[] seq)
+        {
+            // TODO (optimization): rather than vector of boxed int, can we use the interfaces so we can stay primitive array?
+            super(Type.r, key, seq == null ? null : PersistentVector.create(IntStream.of(seq).mapToObj(Integer::valueOf).collect(Collectors.toList())));
+        }
+    }
+
+    private static class Append extends Action
+    {
+        protected Append(int key, int value)
+        {
+            super(Type.append, key, value);
+        }
+    }
+
+    private static class Event extends ObjectPersistentMap
+    {
+        enum Type
+        {
+            invoke, ok, fail; // info is left out as burn test does not have access to the original request, so can't populate an "invoke" event
+
+            final Keyword keyword;
+
+            Type()
+            {
+                keyword = RT.keyword(null, name());
+            }
+        }
+
+        private final int process;
+        private final Event.Type type;
+        private final List<Action> actions;
+        private final long time;
+        private long index = -1;
+
+        private Event(int process, Type type, long time, List<Action> actions)
+        {
+            super(Keys.eventKeys);
+            this.process = process;
+            this.type = type;
+            this.actions = actions;
+            this.time = time;
+        }
+
+        public static Object toClojure(List<Event> events)
+        {
+            return PersistentVector.create(events);
+        }
+
+        @Override
+        public boolean containsKey(Object key)
+        {
+            if (key == Keys.index)
+                return index != -1;
+            return super.containsKey(key);
+        }
+
+        @Override
+        public Object valAt(Object key, Object notFound)
+        {
+            if      (key == Keys.process) return process;
+            else if (key == Keys.index)   return index == -1 ? notFound : index;
+            else if (key == Keys.time)    return time;
+            else if (key == Keys.type)    return type.keyword;
+            else if (key == Keys.value)   return actions;
+            return notFound;
+        }
+
+        @Override
+        public IPersistentMap assoc(Object key, Object val)
+        {
+            if (key == Keys.index)
+                index = ((Long) val).longValue();
+            else
+                throw new UnsupportedOperationException("Unable to update key " + key);
+            return this;
+        }
+    }
+
+    private static class Keys
+    {
+        // event keys
+        private static Keyword process = RT.keyword(null, "process");
```

Comment:
nit: Everything from `process` to `emptyTransactionGraph ` looks like it could be `final`